besu/src/test/java/org/hyperledger/besu/RunnerBuilderTest.java

```diff
@@ -133,8 +144,53 @@ public void enodeUrlShouldHaveAdvertisedHostWhenDiscoveryDisabled() {
             .ipAddress(p2pAdvertisedHost)
             .discoveryPort(0)
             .listeningPort(p2pListenPort)
-            .nodeId(new byte[64])
+            .nodeId(besuController.getNodeKey().getPublicKey().getEncoded())
             .build();
     assertThat(runner.getLocalEnode().orElseThrow()).isEqualTo(expectedEodeURL);
   }
+
+  @Test
+  public void movingAcrossProtocolSpecsUpdatesNodeRecord() {
+    final BlockDataGenerator gen = new BlockDataGenerator();
+    final String p2pAdvertisedHost = "172.0.0.1";
+    final int p2pListenPort = 30301;
+    final StorageProvider storageProvider = new InMemoryKeyValueStorageProvider();
+    final Block genesisBlock = gen.genesisBlock();
+    final MutableBlockchain blockchain =
+        createInMemoryBlockchain(genesisBlock, new MainnetBlockHeaderFunctions());
+    when(besuController.getProtocolContext().getBlockchain()).thenReturn(blockchain);
+    final Runner runner =
+        new RunnerBuilder()
+            .discovery(true)
+            .p2pListenInterface("0.0.0.0")
+            .p2pListenPort(p2pListenPort)
+            .p2pAdvertisedHost(p2pAdvertisedHost)
+            .p2pEnabled(true)
+            .natMethod(NatMethod.NONE)
+            .besuController(besuController)
+            .ethNetworkConfig(mock(EthNetworkConfig.class))
+            .metricsSystem(mock(ObservableMetricsSystem.class))
+            .jsonRpcConfiguration(mock(JsonRpcConfiguration.class))
+            .graphQLConfiguration(mock(GraphQLConfiguration.class))
+            .webSocketConfiguration(mock(WebSocketConfiguration.class))
+            .metricsConfiguration(mock(MetricsConfiguration.class))
+            .vertx(Vertx.vertx())
+            .dataDir(dataDir.getRoot().toPath())
+            .storageProvider(storageProvider)
+            .forkIdSupplier(() -> Collections.singletonList(Bytes.EMPTY))
+            .build();
+    runner.start();
+    when(besuController.getProtocolSchedule().streamMilestoneBlocks()).thenReturn(Stream.of(1L));
+    final Block block =
+        gen.block(
+            BlockDataGenerator.BlockOptions.create()
+                .setBlockNumber(1)
+                .setParentHash(genesisBlock.getHash()));
+    blockchain.appendBlock(block, gen.receipts(block));
+    assertThat(
+            storageProvider
+                .getStorageBySegmentIdentifier(BLOCKCHAIN)
+                .get("local-enr-seqno".getBytes(StandardCharsets.UTF_8)))
+        .isNotNull();
+  }
```

Comment:
is this actually checking that the sequence has incremented or just that it isn't null?  could we assert that it has gone from 0 -> 1 ?

Reply:
I'm really glad you mentioned this because in looking at it I actually found out I had an unrelated test bug (nulls vs optionals). I also expanded it a bit to update multiple times.